frontend/src/components/graph/index.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { Component } from 'react';
+import * as d3 from 'd3';
+import { IPaper } from '../../types/paper';
+import CitationLink from './CitationLink';
+import PaperNode from './PaperNode';
+import './graph.css';
+
+/**
+ * Parameters that are passed from the parent to the PaperGraph.
+ */
+interface IProps {
+  // A list of papers that should be displayed.
+  papers: IPaper[];
+}
+
+/**
+ * The PaperGraph displays a list of papers and their connections.
+ * It uses D3 network graphs.
+ * See also https://www.d3-graph-gallery.com/network.html
+ */
+export default class PaperGraph extends Component<IProps> {
+  private canvas = React.createRef<SVGSVGElement>();
+
+  public render() {
+    // We will paint on this SVG canvas
+    return <svg ref={this.canvas} className="canvas"></svg>;
+  }
+
+  public componentDidMount() {
+    // We got a 'Paper' passed from the props.
+    // To display a 'Paper' on the graph, we need to convert it to a 'PaperNode'
+    const paperNodes = this.props.papers.map((p) => new PaperNode(p));
+
+    // Furthermore, we need to find out which links we want to display between 'PaperNodes'
+    const citationLinks: CitationLink[] = [];
+    for (const p of paperNodes) {
+      for (const c of p.paper.citations) {
+        // Try to find the paper that is cited (this is super slow for large numbers of papers...)
+        const citedPaper = paperNodes.find((n) => n.paper.id === c);
+        if (citedPaper !== undefined) {
```

Comment:
I think it's great that you're already considering papers that reference an "unknown" paper here